packages/api-gateway/src/clients/catchClientError.ts

```diff
@@ -0,0 +1,31 @@
+import { z } from "zod";
+import { ApiError, genericError } from "pagopa-interop-models";
+import { ErrorCodes } from "../models/errors.js";
+
+const CatchableCodes = z.enum(["404", "403"]);
+type CatchableCodes = z.infer<typeof CatchableCodes>;
+const ClientError = z.object({
+  response: z.object({
+    status: z.coerce.string().pipe(CatchableCodes),
+  }),
+});
+type StatusCodeErrorsMap = Record<
+  CatchableCodes,
+  ApiError<ErrorCodes> | undefined
+>;
+export const clientStatusCodeToError = (
+  clientResult: unknown,
+  statusCodesErrorMap: Partial<StatusCodeErrorsMap>
+): Error => {
+  const clientError = ClientError.safeParse(clientResult);
+
+  if (clientError.success) {
+    const { status } = clientError.data.response;
+    const error = statusCodesErrorMap[status];
+    if (error) {
+      return error;
+    }
+  }
+
+  return genericError("Unexpected error receved from API client");
```

Comment:
You're right, I'm not logging the info and am "overriding" the error with this new generic error.
If it is a code not handled in the map I should return the error as it is, it will be rethrown and logged in makeApiProblem which already logs all uncatched errors...
On my way to fix this!

Reply:
Done and tested all cases! If I return the error as it is, then it's caught and logged as expected by the makeApiProblem. I added a log when catching an error instead, and improved the function readability